source/rust.txt

```diff
@@ -54,68 +52,61 @@ corresponding connection code samples.
       different runtime, see
       `Configuring the async runtime <https://github.com/mongodb/mongo-rust-driver#configuring-the-async-runtime>`__.
 
-      .. code-block:: rust
+      .. literalinclude:: /includes/connection-snippets/scram/rust-connection-async.rs
+         :language: rust
 
-         use mongodb::{bson::doc, options::ClientOptions, Client};
+   .. tab:: Sync
+      :tabid: rust-sync
 
-         #[tokio::main]
-         async fn main() -> mongodb::error::Result<()> {
-             // Parse your connection string into an options struct
-             let mut client_options =
-                 ClientOptions::parse("mongodb+srv://<username>:<password>@<cluster-url>/test?w=majority")
-                     .await?;
+      Make sure you enabled the sync API. See
+      `Enabling the sync API <https://github.com/mongodb/mongo-rust-driver#enabling-the-sync-api>`__
+      for more details.
 
-             // Manually set an option
-             client_options.app_name = Some("Rust Demo".to_string());
+      .. literalinclude:: /includes/connection-snippets/scram/rust-connection-sync.rs
+         :language: rust
+      
+This connection snippet uses the {+stable-api+} feature which you can use when
+connecting to MongoDB Server v5.0 and later and the Rust driver v2.0 and 
+later.
 
-             // Get a handle to the cluster
-             let client = Client::with_options(client_options)?;
+When you use this feature, you can update your driver or server without 
+worrying about backward compatibility issues with any commands covered by the
+{+stable-api+}.
 
-             // Ping the server to see if you can connect to the cluster
-             client
-                 .database("admin")
-                 .run_command(doc! {"ping": 1}, None)
-                 .await?;
-             println!("Connected successfully.");
+.. include:: /includes/stable-api-notice.rst
 
-             // List the names of the databases in that cluster
-             for db_name in client.list_database_names(None, None).await? {
-                 println!("{}", db_name);
-             }
-             Ok(())
-         }
+.. _connect-atlas-no-stable-api-php-driver:
 
-   .. tab:: Sync
-      :tabid: rust-sync
+Connect to MongoDB Atlas Without the Stable API
+-----------------------------------------------
 
-      Make sure you enabled the sync API. See
-      `Enabling the sync API <https://github.com/mongodb/mongo-rust-driver#enabling-the-sync-api>`__
-      for more details.
+If you are using a version of MongoDB or the driver that doesn't support the
+{+stable-api+}, you can use the connection snippets in the following
```

Comment:
S: Some specifics might be helpful here if the user didn't carefully read the earlier parts of the page

```suggestion
If you're using a version of the Rust Driver before 2.0 or a version of MongoDB Server before 5.0, you can use the connection snippets in the following
```

(or something similar to this)

Reply:
IMO, phrasing it this way requires the user to infer that these versions don't support the Stable API feature, which I think is more confusing than the current phrasing.